### PR TITLE
fix(levm): wrong up_front_cost calculation

### LINKED ISSUE #1497: LEVM: upfront calculation cost bug
The test `senderBalance.json` is failing because the upfront cost calculation is not taking into account `EIP-1559`

### PR DESCRIPTION
**Pull request by LeanSerra**
**Motivation**

The current implementation of the up_front_cost calculation is wrong, this PR aims to fix that

**Description**

- Keep the calculation we use now for tx validation
- Modify the value we deduct from the sender account to comply with eip-1559

Closes #1497

### CODE REVIEW

File: crates/vm/levm/src/vm.rs [hunk 1]
```diff
@@ -554,6 +549,29 @@ impl VM {
...
+            .checked_mul(self.env.gas_limit)
+            .ok_or(VMError::TxValidation(
+                TxValidationError::GasLimitPriceProductOverflow,
+            ))?;
+
+        let balance_for_valid_tx = gas_fee_for_valid_tx
+            .checked_add(value)
+            .ok_or(InternalError::UndefinedState(1))?
+            .checked_add(blob_gas_cost)
+            .ok_or(InternalError::UndefinedState(1))?;
```
Comment (JereSalo) [inline crates/vm/levm/src/vm.rs:567]: In this case we can return an `InsufficientAccountFunds`. The user won't be able to pay for that sum anyways 😆

**Review by JereSalo — APPROVED** [review]
Looks good! I just left some simple suggestions.
We can merge after addressing those

File: crates/vm/levm/src/vm.rs [hunk 2]
```diff
@@ -554,6 +549,29 @@ impl VM {
         // https://eips.ethereum.org/EIPS/eip-4844
         let blob_gas_cost = self.get_max_blob_gas_cost()?;
 
+        // For the transaction to be valid the sender account has to have a balance >= gas_price * gas_limit + value if tx is type 0 and 1
+        // balance >= max_fee_per_gas * gas_limit + value + blob_gas_cost if tx is type 2
```
[the remark this answers is not in the document]
Comment (LeanSerra) [reply crates/vm/levm/src/vm.rs]: Done in [7e09e08](https://github.com/lambdaclass/ethrex/pull/1504/commits/7e09e08265cc156e967c678e6b31870744665b4b)

File: crates/vm/levm/src/vm.rs (hunk 1 above)
Comment (LeanSerra) [reply crates/vm/levm/src/vm.rs]: Done in [b826efc](https://github.com/lambdaclass/ethrex/pull/1504/commits/b826efcbdb14308362b0e50686a7c7fa6df427a1)

**Pull request merged**